### PR TITLE
Oop basic

### PR DESCRIPTION
**Pull request by MaximGolyak**

### CODE REVIEW

File: src/main/java/core/basesyntax/Bulldozer.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package core.basesyntax;
+
+public class Bulldozer extends Machine {
+    @Override
+    public void doWork() {
+        System.out.println("Bulldozer is starting its work!");
+
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/Bulldozer.java:7]:
```suggestion
```

File: src/main/java/core/basesyntax/Bulldozer.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    }
+
+    @Override
+    public void stopWork() {
+        System.out.println("Bulldozer has stopped working and is refueling.");
+
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/Bulldozer.java:13]:
```suggestion
```

File: src/main/java/core/basesyntax/Excavator.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package core.basesyntax;
+
+public class Excavator extends Machine {
+
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/Excavator.java:4]:
```suggestion
```

File: src/main/java/core/basesyntax/Excavator.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    @Override
+    public void doWork() {
+        System.out.println("Excavator is working - digging a hole!");
+    }
+
+    @Override
+    public void stopWork() {
+        System.out.println("Excavator has stopped working - safe to install the formwork.");
+
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/Excavator.java:13]:
```suggestion
```

File: src/main/java/core/basesyntax/Excavator.java [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+    }
+
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/Excavator.java:15]:
```suggestion
```

File: src/main/java/core/basesyntax/MainApp.java [hunk 6]
```diff
@@ -1,5 +1,17 @@
...
+    public static void main(String[] args) {
+        Machine excavator = new Excavator();
+        Machine bulldozer = new Bulldozer();
+        Machine truck = new Truck();
 
+        Machine[] machines = new Machine[]{excavator, bulldozer, truck};
+        for (Machine machine : machines) {
+            machine.doWork();
+            machine.stopWork();
+            System.out.println("----------------");
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/MainApp.java:13]:
```suggestion
```

File: src/main/java/core/basesyntax/Machine.java [hunk 7]
```diff
@@ -0,0 +1,21 @@
+package core.basesyntax;
+
+public abstract class Machine {
+    private String name;
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/Machine.java:4]: You actually don't use porperty `name` and `get / set / getMachineInfo` methods anywhere, so remove them

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Bulldozer.java (hunk 1 above)
Comment (Elena-Bruyako) [reply src/main/java/core/basesyntax/Bulldozer.java:7]: remove empty line

File: src/main/java/core/basesyntax/Bulldozer.java (hunk 2 above)
Comment (Elena-Bruyako) [reply src/main/java/core/basesyntax/Bulldozer.java:13]: remove empty line

File: src/main/java/core/basesyntax/Excavator.java (hunk 3 above)
Comment (Elena-Bruyako) [reply src/main/java/core/basesyntax/Excavator.java:4]: remove empty line

File: src/main/java/core/basesyntax/Excavator.java (hunk 5 above)
Comment (Elena-Bruyako) [reply src/main/java/core/basesyntax/Excavator.java:15]: remove empty line

File: src/main/java/core/basesyntax/Machine.java [hunk 8]
```diff
@@ -0,0 +1,9 @@
+package core.basesyntax;
+
+public abstract class Machine {
+
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/Machine.java:4]: remove empty line

File: src/main/java/core/basesyntax/MainApp.java (hunk 6 above)
Comment (Elena-Bruyako) [reply src/main/java/core/basesyntax/MainApp.java:10]: remove

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
[Please don't add redundant empty lines to your code.](https://github.com/mate-academy/jv-oop/blob/master/checklist.md#please-dont-add-redundant-empty-lines-to-your-code)

File: src/main/java/core/basesyntax/Excavator.java (hunk 3 above)
Comment (MentorMateAcademy) [reply src/main/java/core/basesyntax/Excavator.java:4]: not fixed

File: src/main/java/core/basesyntax/Excavator.java (hunk 5 above)
Comment (MentorMateAcademy) [reply src/main/java/core/basesyntax/Excavator.java:14]: not fixed

**Review by MentorMateAcademy — CHANGES_REQUESTED** [review]

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
You haven't fixed previous comments. If you have some questions about them or task, please ask in chat

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
one commit needs to be fixed

**Review by Elena-Bruyako — APPROVED** [review]

**Pull request closed without merging**